### ISSUE TITLE
New release?

### ISSUE THREAD

**Issue by reanimus** [salty-horse/ibus-uniemoji#40]
I've got one of my laptops set up to build the latest git commit of ibus-uniemoji, which works fine for that one, but the latest release was 4 years ago, and that's the one that distros have packaged, if they have ibus-uniemoji in the repositories. There have been a lot of updates and improvements since then, especially around shortnames for emoji and other improvements. Any chance of another release to update the stale distro packages?

**Comment by salty-horse** [comment]
Yes. I do have a few worries:
* I haven't received any feedback on the latest changes that include listing text representations alongside emoji. Is it something that people like? Should I remove it?
* EmojiOne (now JoyPixels), which I use for aliases, has moved away from the MIT license to their own [license](https://www.joypixels.com/licenses/free). I don't know if distros will like that.

**Comment by topotech** [comment]
Using [emojitwo](https://github.com/EmojiTwo/emojitwo) wouldn't solve problem 2?

**Comment by salty-horse** [comment]
Didn't know about emojitwo. Their `emoji.json` database hasn't been updated in 4 years, though.

**Comment by mvz** [comment]
Yes, the JoyPixels license is definitely not suitable for, e.g., distribution in Debian.

**Comment by mike-fabian** [comment]
> Yes. I do have a few worries:
> 
>     * I haven't received any feedback on the latest changes that include listing text representations alongside emoji. Is it something that people like? Should I remove it?

I think that is helpful. Depending on the font, it is sometimes difficult to distinguish similar looking emoji, the text representation helps for that and sometimes explains what an emoji means I cannot recognize at all.

> 
>     * EmojiOne (now JoyPixels), which I use for aliases, has moved away from the MIT license to their own [license](https://www.joypixels.com/licenses/free). I don't know if distros will like that.

In ibus-typing-booster,

https://mike-fabian.github.io/ibus-typing-booster/

I do the matching of emoji mostly using the data from the Unicode emoji files

https://www.unicode.org/Public/emoji/14.0/ReadMe.txt
https://www.unicode.org/Public/14.0.0/ucd/emoji/emoji-data.txt
https://www.unicode.org/Public/emoji/14.0/emoji-sequences.txt
https://www.unicode.org/Public/emoji/14.0/emoji-test.txt
https://www.unicode.org/Public/14.0.0/ucd/emoji/emoji-variation-sequences.txt
https://www.unicode.org/Public/emoji/14.0/emoji-zwj-sequences.txt

and the CLDR annotations for emoji:

https://github.com/unicode-org/cldr/tree/main/common/annotations

For aliases like `:-)` I still use an old version of the EmojiOne json file.

Using the Unicode and the CLDR data is more difficult as there is much more data to parse but it also makes it possible to do the matching in many languages as the CLDR data is available in many languages:


https://mike-fabian.github.io/ibus-typing-booster/docs/user/#7_5

**Comment by salty-horse** [comment]
@mike-fabian Do you know of a new source for aliases? For example, I would expect most people to associate the new `U+1FAF0 Hand With Index Finger And Thumb Crossed` with "finger heart".

**Comment by salty-horse** [comment]
I see that JoyPixels now has a [separate MIT license for "non-artwork"](https://github.com/joypixels/emoji-toolkit/blob/master/LICENSE.md). I suspect this means their `emoji.json` file is good to use?

They still haven't updated to 14.0 though :)

**Comment by mike-fabian** [comment]
> I see that JoyPixels now has a [separate MIT license for "non-artwork"](https://github.com/joypixels/emoji-toolkit/blob/master/LICENSE.md). I suspect this means their `emoji.json` file is good to use?
> 
> They still haven't updated to 14.0 though :)

My first impression is that it would be OK to use just the json file.

So maybe we can update to the latest version of that json file.

**Comment by mike-fabian** [comment]
> @mike-fabian Do you know of a new source for aliases? For example, I would expect most people to associate the new `U+1FAF0 Hand With Index Finger And Thumb Crossed` with "finger heart".

https://github.com/unicode-org/cldr/blob/main/common/annotations/en.xml#L774

		<annotation cp="🫰">expensive | hand with index finger and thumb crossed | heart | love | money | snap</annotation> 
		<annotation cp="🫰" type="tts">hand with index finger and thumb crossed</annotation>

So if you match on “finger” and “heart” in this, it should work.

**Comment by carlocastoldi** [comment]
I see that with c9d758fd40b74ba043b1ef89a11b858fc08f65b1 you clarified on the license used by ibus-uniemoji. Is that a good start for a release? :D

**Comment by salty-horse** [comment]
That commit also moved some files around, and I would really like confirmation that the installation/uninstallation process works fine.

I just pushed an update with the new JoyPixels file released last month.

**Comment by carlocastoldi** [comment]
>That commit also moved some files around, and I would really like confirmation that the installation/uninstallation process works fine.

is [`ibus-uniemoji-git`](https://aur.archlinux.org/packages/ibus-uniemoji-git) from AUR a good test for that? Because I built it today and it worked flawlessly!
